### PR TITLE
Fix legend marker spacing

### PR DESCRIPTION
**Pull request by yp0413150120**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Description of change

1、修复`legend`组件中`item.marker.spacing`不生效
2、添加相关测试用例

### CODE REVIEW

File: src/legend/category.ts [hunk 1]
```diff
@@ -696,13 +702,13 @@ class Category extends LegendBase<CategoryLegendCfg> implements IList {
     const translate =
       layout === 'horizontal'
         ? {
-            x: 0,
-            y: pageHeight * (1 - currentPageIndex),
-          }
+          x: 0,
```
Comment (visiky) [inline src/legend/category.ts:705]: 注意下：格式化的改动是否合理? 这里似乎不需要？

File: src/legend/category.ts [hunk 2]
```diff
@@ -414,7 +414,13 @@ class Category extends LegendBase<CategoryLegendCfg> implements IList {
     let curX = 0; // 记录当前 x 的位置
     if (marker) {
       const markerShape = this.drawMarker(subGroup, marker, item, itemHeight);
-      curX = markerShape.getBBox().maxX + marker.spacing;
+      const itemMarkerSpacing = get(item, ['marker', 'spacing']);
+      if (isNumber(itemMarkerSpacing) && itemMarkerSpacing !== marker.spacing) {
+        // 如果 item 的特异配置 marker.spacing 和 通用配置 marker.spacing 不一致，采用 item 的配置
+        curX = markerShape.getBBox().maxX + itemMarkerSpacing;
+      } else {
+        curX = markerShape.getBBox().maxX + marker.spacing;
```
Comment (visiky) [inline src/legend/category.ts:422]: 简化下：
```ts
let spacing = marker.spacing;
if (isNumber(get(item, ['marker', 'spacing']))) {
   spacing = get(item, ['marker', 'spacing']);
}

curX = markerShape.getBBox().maxX + spacing;
```

File: src/legend/category.ts (hunk 1 above)
Comment (yp0413150120) [reply src/legend/category.ts]: 开启了vs保存自动格式化，已经恢复了

File: src/legend/category.ts (hunk 2 above)
Comment (yp0413150120) [reply src/legend/category.ts]: 已修改

**Review by visiky — APPROVED** [review]

**Comment by visiky** [comment]
close: https://github.com/antvis/G2Plot/issues/2671

**Pull request merged**